### PR TITLE
H2 mpl

### PR DESCRIPTION
**Pull request by Scinopode**
This replaces all former methods for assigning material properties with the more general MPL methods. Tests are updated accordingly.

1. [x] Feature description was added to the [changelog](https://github.com/ufz/ogs/wiki/Release-notes-6.2.2)
2. [x] Tests covering your feature were added?
3. [x] Any new feature or behavior change was documented?

### CODE REVIEW

File: MaterialLib/MPL/Properties/SaturationLiakopoulos.h [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    Medium* _medium;
+    /**
+  Parameters for Liakopoulos saturation curve taken from:
+  Asadi, R., Ataie-Ashtiani, B. (2015): A Comparison of finite volume
+  formulations and coupling strategies for two-phase flow in deforming
+  porous media. Comput. Geosci., p. 24ff.
+
+  Those parameters are fixed for that particular model, no need to change them.
+*/
+    const double _residual_liquid_saturation = 0.2;
```
Comment (endJunction) [inline MaterialLib/MPL/Properties/SaturationLiakopoulos.h]:
```suggestion
    constexpr double _residual_liquid_saturation = 0.2;
```

File: Tests/Data/Parabolic/TwoPhaseFlowPP/McWhorter/TwoPhase_mcwt_line.prj [hunk 2]
```diff
@@ -84,6 +24,82 @@
...
+               <entry_pressure>5000</entry_pressure>
+            </property>
+
+            <property>
+               <name>relative_permeability</name>
+               <type>RelPermBrooksCorey</type>
+               <residual_liquid_saturation>0.0</residual_liquid_saturation>
+               <residual_gas_saturation>0.0</residual_gas_saturation>
+               <lambda>2</lambda>
+               <min_relative_permeability_liquid>0</k_rel_min_liquid>
```
Comment (endJunction) [inline Tests/Data/Parabolic/TwoPhaseFlowPP/McWhorter/TwoPhase_mcwt_line.prj]:
```suggestion
               <min_relative_permeability_liquid>0</min_relative_permeability_liquid>
```

File: Tests/Data/Parabolic/TwoPhaseFlowPP/McWhorter/TwoPhase_mcwt_line.prj [hunk 3]
```diff
@@ -84,6 +24,82 @@
...
+               <min_relative_permeability_gas>1e-9</k_rel_min_gas>
```
Comment (endJunction) [inline Tests/Data/Parabolic/TwoPhaseFlowPP/McWhorter/TwoPhase_mcwt_line.prj]:
```suggestion
               <min_relative_permeability_gas>1e-9</min_relative_permeability_gas>
```

File: ProcessLib/TwoPhaseFlowWithPP/TwoPhaseFlowWithPPLocalAssembler-impl.h [hunk 4]
```diff
@@ -122,27 +109,67 @@ void TwoPhaseFlowWithPPLocalAssembler<
...
+
+        auto const lambda_wet = k_rel_wet / mu_wet;
+
+        auto const perm = medium.property(MPL::PropertyType::permeability)
+                              .template value<double>(variables, pos, t);
+
+        GlobalDimMatrixType permeability = GlobalDimMatrixType::Zero(
+            _element.getDimension(), _element.getDimension());
+
+        permeability.diagonal().setConstant(perm);
```
Comment (renchao-lu) [inline ProcessLib/TwoPhaseFlowWithPP/TwoPhaseFlowWithPPLocalAssembler-impl.h]: i think we could call the function `MaterialPropertyLib::formEigenTensor` here
`auto const& K = MaterialPropertyLib::formEigenTensor<GlobalDim>(              medium.property(MaterialPropertyLib::PropertyType::permeability).value(vars, pos, t));`

File: Tests/MaterialLib/TestMPLSaturationBrooksCorey.cpp [hunk 5]
```diff
@@ -49,30 +49,15 @@ TEST(MaterialPropertyLib, SaturationBrooksCorey)
...
     {
         variable_array[static_cast<int>(
             MaterialPropertyLib::Variable::capillary_pressure)] = p_cap;
 
-        s_L = medium->property(MaterialPropertyLib::PropertyType::saturation)
-                  .template value<double>(variable_array, pos, time);
-        ds_L_dp_cap =
+        auto s_L =
+            medium->property(MaterialPropertyLib::PropertyType::saturation)
+                .template value<double>(variable_array, pos, time);
```
Comment (renchao-lu) [inline Tests/MaterialLib/TestMPLSaturationBrooksCorey.cpp]: It would be better to change `time` with `t` for code consistency.

**Pull request merged**